### PR TITLE
REF - Use function to get component name from permission

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Refactors some long-winded code which was attempting to do what an existing function does better

Before
----------------------------------------
Try to guess component name from permission name, based on string fudgery

After
----------------------------------------
Use a function which does it without guessing

Comments
----------------------------------------
I'm fairly certain this change is correct, but someone else should look at it to ensure I'm not missing some important difference between the existing code (silly as it was) and the new function.

### CODE REVIEW

**Comment by colemanw** [comment]
Retest this please

**Comment by demeritcowboy** [comment]
I can take a look.

**Comment by demeritcowboy** [comment]
Looks ok.

One sanity check I did was to compare the output of `cv ev --user=something "$perms = CRM_Core_Permission::basicPermissions(); foreach ($perms as $p => $dontcare) { var_dump(CRM_Core_BAO_Dashboard::checkPermission([$p], null)); }"` with admin/regular/anonymous users. Checks out.

This function also appears to only be called from one place in universe (the same file), so it limits the scope. I'm ok to merge just might be worth making the function private.

File: CRM/Core/BAO/Dashboard.php [hunk 1]
```diff
@@ -189,17 +176,10 @@ public static function checkPermission($permissions, $operator) {
         }
       }
 
-      if (!$showDashlet && !$hasPermission) {
-        return FALSE;
-      }
```
Comment (colemanw) [inline CRM/Core/BAO/Dashboard.php:194]: Wow, a triple negative! I wonder if there's a simpler way to write that?

**Comment by colemanw** [comment]
Ok @demeritcowboy I changed the function to be private and added some additional cleanup for your reviewing pleasure :)

File: CRM/Core/BAO/Dashboard.php (hunk 1 above)
Comment (demeritcowboy) [reply CRM/Core/BAO/Dashboard.php:194]: Hehe. Seems there isn't not a way to avoid it.

Comment (colemanw) [reply CRM/Core/BAO/Dashboard.php:194]: I wouldn't be unsurprised if we couldn't simplify it.

**Pull request merged**